BirtherDay/Common/Manager/BDNavigationRoutingView.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  BDNavigationRoutingView.swift
+//  BirtherDay
+//
+//  Created by Soop on 6/5/25.
+//
+
+import SwiftUI
+
+struct BDNavigationRoutingView: View {
+    @State var destination: BDAppPath
+    
+    // 뷰모델 주입
+    let createCouponViewModel: CreateCouponViewModel
+    let couponDetailViewModel: CouponDetailViewModel
```

Comment:
p3

각 view Model이 서로 View에 영향이 가고, UI에 영향을 끼칠 경우 @ObservedObject를 넣어야 하는뎅
숲이 의도한 것은, 각각의 View가 UI에 영향을 주지 않아서 넣지 않으신 것 같네요! 

혹시 이렇게 할 경우 아래 상황에서 어떻게 처리되나요?

ex) 사용자가 쿠폰에 들어갈 편지 내용을 작성하고 있음, 중간에 잠깐 템플릿을 변경하고 싶어, 뒤로가기 버튼을 눌러 템플릿을 다른 폼으로 변경함. 이때 다시 letter view로 사용자가 이동했을 때 템플릿에 맞게끔 뒷 배경색이 변경되나요?

```suggestion
    @ObservedObject let createCouponViewModel: CreateCouponViewModel
    @ObservedObject let couponDetailViewModel: CouponDetailViewModel
```

Reply:
새벽에 급하게 수정했더니 그걸 신경 못 썻네요...
CouponDetailViewModel은 @Observable 매크로가 있어서 따로 @ObservedObject를 안 붙여도 됩니다! CreateCouponViewModel은 수정해뒀어요 ~